### PR TITLE
Coffin healing now updates vampire's overall health

### LINKED ISSUE #26324: Wooden coffin does not correctly heal vampires (Breaks health bar and speed)
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

Latest (515.1642)

### Issue Description

I slept in a wooden coffin to heal myself (Uncounted amount of brute + burn UPDATE: No cellular damages( cellular damage from space exposure) ). The health doll became fully green after a few naps and I was not able to find any signs of me being damaged from checking my own injuries / inspecting myself / using a health analyzer BUT I was still slow as if I was damaged and my health bar stayed in it's damaged state.

### What did you expect to happen?

For me to be healed and able to run at full speed with my health bar's state being back to 100% green.

### What happened instead?

My health bar stayed orange and I kept being slow even while running.

### Why is this bad/What are the consequences?

Wood coffins apparently breaks the health of the vampires sleeping in them.. Wich is very bad as the admin I asked for help had to aheal me in order to "fix" the problem.

### Steps to reproduce the issue.

Get damaged (Brute & Burn) by space between cyberiad's maintenance and perma's garden as vampire.
Fulton to lavaland's mining outpost directly from the space explained above.
Sleep in a coffin until health doll is fully healed.
Stay "damaged".

### When did the problem start happening?

Round 41236

### Extra information

_No response_

### Relevant log output/runtime error

_No response_

### PR DESCRIPTION
**Pull request by Migratingcocofruit**
…aling.

## What Does This PR Do
Fixes #26324
Uses human scoped damage adjust procs when healing a vampire via sleeping in a coffin.

## Why It's Good For The Game
Vampires heal properly now

## Testing
-Got robotic leg, became vampire, damaged myself in various ways, went to sleep. 
-Damage was updates and the robotic limb did not heal

## Changelog
:cl:
fix: Coffing healing updates vampire health
/:cl:

### CODE REVIEW

**Review by Burzah — COMMENTED** [review]
I believe this will close out this issue here.
https://github.com/ParadiseSS13/Paradise/issues/26324

**Comment by Migratingcocofruit** [comment]
> I believe this will close out this issue here. #26324

Oh right I forgot to mention it

**Review by Sirryan2002 — APPROVED** [review]

File: code/datums/status_effects/debuffs.dm [hunk 1]
```diff
@@ -654,14 +654,15 @@
...
 	if(dreamer.mind?.has_antag_datum(/datum/antagonist/vampire))
-		if(istype(dreamer.loc, /obj/structure/closet/coffin))
-			dreamer.adjustBruteLoss(-1, FALSE)
-			dreamer.adjustFireLoss(-1, FALSE)
-			dreamer.adjustToxLoss(-1)
-			dreamer.adjustOxyLoss(-1)
-			dreamer.adjustCloneLoss(-0.5)
-			if(dreamer.HasDisease(/datum/disease/critical/heart_failure) && prob(25))
-				for(var/datum/disease/critical/heart_failure/HF in dreamer.viruses)
+		var/mob/living/carbon/human/V = owner
```
Comment (DGamerL) [inline code/datums/status_effects/debuffs.dm:657]: I dislike the single-letter-variable here but it's small in scope, so good for now.
Please use longer variable names in the future

**Review by DGamerL — APPROVED** [review]

File: code/datums/status_effects/debuffs.dm (hunk 1 above)
Comment (Migratingcocofruit) [reply code/datums/status_effects/debuffs.dm:657]: will do.

**Pull request merged**